src/heretic/main.py

```diff
@@ -435,6 +537,8 @@ def objective_wrapper(trial: Trial) -> tuple[float, float]:
             trial.user_attrs["parameters"],
         )
 
+        merged_model_cache = None
```

Comment:
Do we need this? Seems like quite a lot of complexity, and the only upside I can see is that it's a little faster when the user wants to *both* save locally and upload to HF.